doc/src/build/programming-with-objects/ch1-object-basics.md

```diff
@@ -0,0 +1,111 @@
+```
+struct Color {
+    red: u8,
+    green: u8,
+    blue: u8,
+}
+```
+The above `struct` defines a data structure that can represents RGB color. Structures like this can be useful to organize data with complicated semantics. However, instances of structs like `Color` are not Sui objects yet.
+To define a struct that represents a Sui object type, we must add a `key` capability to the definition, and the first field of the struct must be the `id` of the object with type `VersionedID` from the [ID library](../../../../sui_programmability/framework/sources/ID.move):
+```
+use Sui::ID::VersionedID;
+
+struct ColorObject has key {
+    id: VersionedID,
+    red: u8,
+    green: u8,
+    blue: u8,
+}
+```
+Now `ColorObject` represents a Sui object type and can be used to create Sui objects that can be eventually stored on the Sui chain.
+> :books: The [key ability](https://github.com/diem/move/blob/main/language/documentation/book/src/abilities.md#key) in core Move is used to indicate a type can be used as a key for global storage operations. This is similar here in Sui Move, that a struct with key ability can be eventually stored in the global storage. The difference is rather implementation details: Sui's global storage is outside of Move, unlike core Move which manages storage inside Move.
+
+> :bulb: The `VersionedID` type is internal to Sui and most likely you won't need to deal with it directly. For the curious readers, it contains the unique `ID` of the object and the version of the object. Eech time a mutable object is used in a transaction, its version will increment by 1.
+
+### Create Sui Object
+Now that we have learned how to define a Sui object type, how do we create/instantiate an Sui object? In order to create a new Sui object from its type, we must assign an initial value to each of the fields, including `id`. The only way to create a new unique `VersionedID` for a Sui object is to call `TxContext::new_id`. The `new_id` function takes the current transaction context as argument in order to generate unique IDs. The transaction context is of type `&mut TxContext` and should be passed down from an entry function (An [entry function](../move.md#entry-functions) is a function that can be called directly from a transaction.). Let's look at how we may define a constructor for `ColorObject`:
+```
+/// TxContext::Self represents the TxContext module, which allows us call
+/// functions in the module, such as the `sender` function.
+/// TxContext::TxContext represents the TxContext struct in TxContext module.
+use Sui::TxContext::{Self, TxContext};
+
+fun new_color_object(red: u8, green: u8, blue: u8, ctx: &mut TxContext): ColorObject {
+    ColorObject {
+        id: TxContext::new_id(ctx),
+        red,
+        green,
+        blue,
+    }
+}
+```
+> :bulb: Similar to Rust, Move allows us to skip the field values if the field name happens to be the same as the name of the value variable.
+
+### Store Sui Object
```

Comment:
I think the core Move / Sui Move comparison is useful, but perhaps a bit distracting. Maybe it could go in a bulb like the following:

In core Move, we would call `move_to<T>(a: address, t: T)` to add the entry `(a, T) -> t` to the global storage. But because (as explained above) the schema of Sui Move's global storage is different, we use the `Transfer` API's instead of `move_to` or the other [global storage operators]([type-indexed global storage](https://github.com/diem/move/blob/main/language/documentation/book/src/global-storage-operators.md) of core Move. These operators cannot be used in Sui Move.